spec.go

```diff
@@ -225,6 +225,60 @@ type SourceBuild struct {
 	Args map[string]string `yaml:"args,omitempty" json:"args,omitempty"`
 }
 
+// SourceInlineContent is used to specify the content of an inline source.
+type SourceInlineFile struct {
+	// Content is the contents.
+	Contents string `yaml:"content,omitempty" json:"content,omitempty"`
+	// Permissions is the octal file permissions to set on the file.
+	Permissions fs.FileMode `yaml:"permissions,omitempty" json:"permissions,omitempty"`
+	// UID is the user ID to set on the directory and all files and directories within it.
+	// UID must be greater than 0
+	UID int `yaml:"uid,omitempty" json:"uid,omitempty"`
+	// GID is the group ID to set on the directory and all files and directories within it.
+	// GID must be greater than 0
+	GID int `yaml:"gid,omitempty" json:"gid,omitempty"`
+}
+
+// SourceInlineDir is used by by [SourceInline] to represent a filesystem directory.
+type SourceInlineDir struct {
+	// Files is the list of files to include in the directory.
+	// The map key is the name of the file.
+	//
+	// Files with path separators in the key will be rejected.
+	//
+	// Files are created after directories are created.
+	Files map[string]*SourceInlineFile `yaml:"files,omitempty" json:"files,omitempty"`
+	// Dirs is the list of directories to include in the directory.
+	// The map key is the name of the directory.
+	//
+	// Directories with path separators in the key will be rejected.
+	//
+	// Directories are created before files are created.
+	Dirs map[string]*SourceInlineDir `yaml:"dirs,omitempty" json:"dirs,omitempty"`
+	// Permissions is the octal permissions to set on the directory.
+	Permissions fs.FileMode `yaml:"permissions,omitempty" json:"permissions,omitempty"`
+
+	// UID is the user ID to set on the directory and all files and directories within it.
+	// UID must be greater than 0
+	UID int `yaml:"uid,omitempty" json:"uid,omitempty"`
```

Comment:
Same as above, just making sure comment matches validation

Reply:
Nice catch. I think this was copilot :)
It tries hard.